### PR TITLE
[public-api] Route api.<domain> to public api service & deployment

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

* Depends on https://github.com/gitpod-io/gitpod/pull/9362
* Part of https://github.com/gitpod-io/gitpod/issues/9229

## How to test

Navigate to preview build. Prepend `api.` to the URL and you should see `hello world`

https://api.mp-papi-caddy.staging.gitpod-dev.com

In prod, we haven't configured the DNS record to point to our loadbalancer so it won't work, yet. This is fine, we can iterate in preview for now.

## Release Notes

```release-note
NONE
```

## Documentation

NONE

/hold

### CODE REVIEW

**Comment by roboquat** [comment]
@easyCZ: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/public-api-server/main.go [hunk 1]
```diff
@@ -22,6 +22,10 @@ func main() {
 		logger.WithError(err).Fatal("Failed to initialize public api server.")
 	}
 
+	if err := register(srv); err != nil {
+		logger.WithError(err).Fatal("Failed to register services.")
+	}
+
```
Comment (andrew-farries) [inline components/public-api-server/main.go:28]: Was this change necessary to set up the routing or just overlooked in earlier PRs?

**Review by andrew-farries — APPROVED** [review]

File: components/public-api-server/main.go (hunk 1 above)
Comment (easyCZ) [reply components/public-api-server/main.go:28]: Overlooked from other PRs.
